test/lsptoolshost/integrationTests/index.ts

```diff
@@ -3,11 +3,73 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as vscode from 'vscode';
+import * as fsExtra from 'fs-extra';
+import path from 'path';
+import { AggregatedResult } from '@jest/test-result';
 import { runIntegrationTests } from '../../runIntegrationTests';
 import { jestIntegrationTestProjectName } from './jest.config';
+import { activateCSharpExtension } from './integrationHelpers';
 
 export async function run() {
     process.env.RUNNING_INTEGRATION_TESTS = 'true';
 
-    await runIntegrationTests(jestIntegrationTestProjectName);
+    await activateCSharpExtension();
+    await moveLogs('activated');
+
+    let anyFailures = false;
+    let results: AggregatedResult | undefined;
+    if (process.env.TEST_FILE_FILTER) {
+        results = await runIntegrationTests(jestIntegrationTestProjectName);
+        await moveLogs(path.basename(process.env.TEST_FILE_FILTER, '.integration.test.ts'));
+        anyFailures = anyFailures || !results.success;
+    } else {
+        const workingDirectory = process.cwd();
+
+        const testFiles = (await fsExtra.readdir(__dirname))
+            .filter((file) => file.endsWith('.integration.test.js'))
```

Comment:
instead of doing this, I think you could probably simplify all this by getting the [rootPath](https://github.com/dotnet/vscode-csharp/blob/main/tasks/projectPaths.ts#L10) here, and then just go directly to the typescript files?